src/index.test.ts

```diff
@@ -129,15 +125,12 @@ function testExport(
                 },
               },
             });
-            expect(onError).toHaveBeenCalledTimes(1);
+            expect(onError).toHaveBeenCalled();
```

Comment:
can't control how often `useSyncExternalStore` is called with the same granularity that you can a `useEffect`, so the `onError`'s end up running more often. I still think the trade-offs are worth it, since no `useEvent`-like hooks are necessary anymore, and this package does really seem like a textbook case of when to use `useSyncExternalStore`